src/gsi_ncdiag/ncdiag_to_feedback.py

```diff
@@ -43,14 +43,22 @@ def main(oman, ombg, output_file, output_dir='./'):
 
     # this is specific now to radiances
     g.copy(g['GsiHofXBc'], k, 'oman')
-    k['oman']['brightnessTemperature'][:, :] = k['ObsValue']['brightnessTemperature'][:, :] - k['hofx1']['brightnessTemperature'][:, :]
-
     f.copy(f['GsiHofXBc'], k, 'ombg')
-    k['ombg']['brightnessTemperature'][:, :] = k['ObsValue']['brightnessTemperature'][:, :] - k['hofx0']['brightnessTemperature'][:, :]
-
-    # ObsBias from analysis
-    g.copy(g['GsiBc'], k, 'ObsBias1')
-    k.copy(k['ObsBias1'], k, 'ObsBias0')
+    # there has got to be a better way
+    for var in list(k['ObsValue'].keys()):
+        if k['ObsValue'][var].ndim == 2:
+            k['oman'][var][:, :] = k['ObsValue'][var][:, :] - k['hofx1'][var][:, :]
+            k['ombg'][var][:, :] = k['ObsValue'][var][:, :] - k['hofx0'][var][:, :]
+        elif k['ObsValue'][var].ndim == 1:
+            k['oman'][var][:] = k['ObsValue'][var][:] - k['hofx1'][var][:]
+            k['ombg'][var][:] = k['ObsValue'][var][:] - k['hofx0'][var][:]
+        else:
+            print(' ... need to handle n-dimension case where n is:', {k['ObsValue'][var].ndim})
```

Comment:
```
        ind_str = k['ObsValue'][var].ndim*':,'
        exec("k['oman'][var]["+ind_str+"] = k['ObsValue'][var]["+ind_str+"] - k['hofx1'][var]["+ind_str+"]")
```

This could replace lines 49 to 56 if you're not sure about how many dimensions may need indexing. 
It inserts a ':,' for the number of dimensions.